test/machines/machines_test.go

```diff
@@ -279,7 +281,8 @@ var _ = framework.SigKubeDescribe("Machines", func() {
 				masterMachine.Annotations = map[string]string{}
 			}
 			masterMachine.Annotations[machinecontroller.ExcludeNodeDrainingAnnotation] = ""
-			f.CreateMachineAndWait(masterMachine, acw)
+			masterMachine = f.CreateMachineAndWait(masterMachine, acw)
```

Comment:
I would put a check for `ProviderID` field being set here rather than inside `f.CreateMachineAndWait`.

Reply:
handled outside of f.CreateMachineAndWait now.